main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func Remove() {
+	var entries []Entry
+	const phonebookFile = "phonebook.json"
+	var id uint32
+	var choise int
+	f, err := os.Open(phonebookFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Print("Enter ID: ")
+	fmt.Scan(&id)
+	for i, v := range entries {
+		if v.ID == id {
+			fmt.Printf("Do you really want to deleat %q %q. Enter <1> to confirm deletion: ", v.FirstName, v.LastName)
+			fmt.Scan(&choise)
+			if choise == 1 {
+				entries = append(entries[:i], entries[i+1:]...)
+				file, err := os.Create(phonebookFile)
+				if err != nil {
+					log.Fatal(err)
+				}
+				defer file.Close()
+				w := bufio.NewWriter(file)
+				defer w.Flush()
+				if err := json.NewEncoder(w).Encode(entries); err != nil {
+					log.Fatal(err)
+				}
+
+			}
+			return
+		}
+	}
+}
+
+func Add() {
+	var entries []Entry
+	var add Entry
+	const phonebookFile = "phonebook.json"
+	scanner := bufio.NewScanner(os.Stdin)
+	f, err := os.Open(phonebookFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	ID := uint32(len(entries))
```

Comment:
This isn't enough. imagine the following scenario:
- add 4 entries (IDs 0, 1, 2, 3 / len 4)
- remove w/ ID 1 (0 2 3 / len 3)
- add new (0 2 3 3 / len 4)
- remove w/ ID 2 (0 3 3 / len 3)
- add new (0 3 3 3 / len 4)
Do you see my point?